scripts/hypothesistooling.py

```diff
@@ -185,3 +199,123 @@ def modified_files():
+    with open(RELEASE_FILE) as i:
+        release_contents = i.read()
+
+    release_lines = release_contents.split('\n')
+
+    m = RELEASE_TYPE.match(release_lines[0])
+    if m is not None:
+        release_type = m.group(1)
+        if release_type not in VALID_RELEASE_TYPES:
+            print('Unrecognised release type %r' % (release_type,))
+            sys.exit(1)
+        del release_lines[0]
+        release_contents = '\n'.join(release_lines).strip()
+    else:
+        print(
+            'RELEASE.rst does not start by specifying release type. The first '
+            'line of the file should be RELEASE_TYPE: followed by one of '
+            'major, minor, or patch, to specify the type of release that '
+            'this is (i.e. which version number to increment). Instead the '
+            'first line was %r' % (release_lines[0],)
+        )
+        sys.exit(1)
+
+    return release_type, release_contents
+
+
+def update_for_pending_release():
+    global __version_info__
+
+    with open(CHANGELOG_FILE) as i:
+        contents = i.read()
+    assert '\r' not in contents
+    lines = contents.split('\n')
+    assert contents == '\n'.join(lines)
+    for i, l in enumerate(lines):
+        if CHANGELOG_BORDER.match(l):
+            assert CHANGELOG_HEADER.match(lines[i + 1]), repr(lines[i + 1])
+            assert CHANGELOG_BORDER.match(lines[i + 2]), repr(lines[i + 2])
+            beginning = '\n'.join(lines[:i])
+            rest = '\n'.join(lines[i:])
+            assert '\n'.join((beginning, rest)) == contents
+            break
+
+    release_type, release_contents = parse_release_file()
+
+    new_version = list(__version_info__)
+    bump = VALID_RELEASE_TYPES.index(release_type)
+    new_version[bump] += 1
+    for i in range(bump + 1, len(new_version)):
+        new_version[i] = 0
+    new_version = tuple(new_version)
+
+    __version_info__ = new_version
+
+    with open(VERSION_FILE) as i:
+        version_lines = i.read().split('\n')
+
+    for i, l in enumerate(version_lines):
+        if 'version_info' in l:
+            version_lines[i] = '__version_info__ = %r' % (new_version,)
+            break
+
+    with open(VERSION_FILE, 'w') as o:
+        o.write('\n'.join(version_lines))
+
+    new_version_string = '.'.join(map(str, new_version))
+
+    now = datetime.utcnow()
+
+    date = max([
+        d.strftime('%Y-%m-%d') for d in (now, now + timedelta(hours=1))
+    ])
+
+    heading_for_new_version = ' - '.join((new_version_string, date))
+    border_for_new_version = '-' * len(heading_for_new_version)
+
+    new_changelog_parts = [
+        beginning.strip(),
+        '',
+        border_for_new_version,
+        heading_for_new_version,
+        border_for_new_version,
+        '',
+        release_contents,
+        '',
+        rest
+    ]
+
+    with open(CHANGELOG_FILE, 'w') as o:
+        o.write('\n'.join(new_changelog_parts))
+
+    git('rm', RELEASE_FILE)
+    git('add', CHANGELOG_FILE, VERSION_FILE)
```

Comment:
Ah, I should have spotted this in review - we need to `git add` the release file too!

@DRMacIver, can you push that change and remove the current release file?